sdk/tests/core/test_az_url_encode.c

```diff
@@ -84,6 +85,42 @@ static void test_url_encode_basic(void** state)
     assert_true(az_span_is_content_equal(
         AZ_SPAN_FROM_BUFFER(buf20), AZ_SPAN_FROM_STR("aBc%2Fg*************")));
   }
+  {
+    int32_t url_length = _az_span_url_encode_calc_length(AZ_SPAN_FROM_STR("aBc/g"));
+    assert_int_equal(url_length, sizeof("aBc%2Fg") - 1);
+  }
+  {
+    // Empty span
+    int32_t url_length = _az_span_url_encode_calc_length(AZ_SPAN_NULL);
```

Comment:
I was thinking of adding these next to each of the `_az_span_url_encode` calls in the tests, like here:
https://github.com/Azure/azure-sdk-for-c/blob/ce4d71bbd0f8a49ab79a47a5f23222bb659773c6/sdk/tests/core/test_az_url_encode.c#L33
https://github.com/Azure/azure-sdk-for-c/blob/ce4d71bbd0f8a49ab79a47a5f23222bb659773c6/sdk/tests/core/test_az_url_encode.c#L82
https://github.com/Azure/azure-sdk-for-c/blob/ce4d71bbd0f8a49ab79a47a5f23222bb659773c6/sdk/tests/core/test_az_url_encode.c#L152


But this is fine too.